scripts/get-user.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+"""Outputs details of a random active unlocked user with a given role.
+
+For supplier users, providing additional <framework> and <lot> arguments allows
+filtering by suppliers who have published services on the given framewok and lot.
+
+Usage:
+    get-user.py <role> [<framework>] [<lot>] [options]
+
+    --api-url=<api_url>  API URL [default: http://localhost:5000]
+    --api-token=<api_token>  API token [default: myToken]
+
+Example:
+    ./get-user.py supplier g9 cloud-hosting
+    ./get-user.py --api-url=localhost:5000 --api-token=myToken admin
+    ./get-user.py --api-url=localhost:5000 --api-token=myToken supplier g-cloud-9
+    ./get-user.py --api-url=localhost:5000 --api-token=myToken supplier dos2 digital-outcomes
+"""
+
+from __future__ import print_function
+
+import random
+import re
+import sys
+
+import dmapiclient
+from docopt import docopt
+
+sys.path.insert(0, '.')  # noqa
+
+
+def get_full_framework_slug(framework):
+    iteration = re.search('(\d+)', framework)
+    if framework.lower().startswith('g'):
+        prefix = 'g-cloud'
+    elif framework.lower().startswith('d'):
+        prefix = 'digital-outcomes-and-specialists'
+    else:
+        return framework
+
+    if iteration:
+        return "{}-{}".format(prefix, iteration.group(1))
+    else:
+        return prefix
+
+
+def get_supplier_id(api_client, framework, lot):
+    service = random.choice([
+        s for s in api_client.find_services(framework=framework, lot=lot)['services']
+        if s['status'] in ['published']
+    ])
+
+    return service['supplierId']
+
+
+def get_random_user(api_client, role, supplier_id=None):
+    return random.choice([
+        u for u in api_client.find_users(role=role, supplier_id=supplier_id)['users']
+        if u['active'] and not u['locked']
+    ])
+
+
+def get_user(api_url, api_token, role, framework, lot):
+    api_client = dmapiclient.DataAPIClient(api_url, api_token)
+
+    if role == 'supplier':
+        framework = get_full_framework_slug(framework)
+        print('Framework: {}'.format(framework))
+        supplier_id = get_supplier_id(api_client, framework, lot)
+        print('Supplier id: {}'.format(supplier_id))
```

Comment:
Printing out framework+supplier id, but no the lot (if given). Be nice to spit everything back out just as confirmation.